### DISCUSSION TITLE
Do not select a file, submit it directly, and report a strange error:multipart.multipart:Did not find CR at end of boundary

### DISCUSSION THREAD

**Discussion by goophps** [fastapi/fastapi#9757] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, UploadFile

app = FastAPI()

@app.post("/sources_files")
async def sources_files(files: list[UploadFile]):
    return 'a'
```

### Description

In the interface document, do not select a file, just click execute

command line errors：
Did not find CR at end of boundary (40)
INFO:     192.168.56.1:58198 - "POST /sources_files HTTP/1.1" 400 Bad Request

interface documentation errors:
400 
Error: Bad Request
{
"detail": "There was an error parsing the body"
}

### Operating System

Linux

### FastAPI Version

0.98.0

### Python Version

3.10.11

**Comment by ersain** [comment]
I had the same issue when I was making a request with a trailing slash at the end, when I removed the slash, I no longer had the problem

**Comment by DukeOfRI** [comment, marked as answer]
I had the same problem. It is caused by a missing `name=` in your html form. In the example below, if you miss `name=...` in your form in `index.html` then you get the error: Did not find CR at end of boundary (40). Simply adding `name=...` will resolve the issue. Note that the value of `name=...` should be equal to the name of the input argument in the upload endpoint it refers to.

Example **app.py**:
```
import uvicorn
from fastapi import FastAPI, Request, UploadFile
from fastapi.responses import HTMLResponse
from fastapi.templating import Jinja2Templates

app = FastAPI()

templates = Jinja2Templates(directory="templates")


@app.get("/", response_class=HTMLResponse)
def get_index(request: Request):
    return templates.TemplateResponse("index.html", {"request": request})


@app.post("/upload")
def upload(request: Request, input_file: UploadFile):
    # Put your logic here. Simply extract file name for now.
    return templates.TemplateResponse(
        "upload.html", {"request": request, "file_name": input_file.filename}
    )


uvicorn.run(app)
```

Example **index.html** in the folder templates:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Upload file</title>
</head>
<body>
    <form action="upload" enctype="multipart/form-data" method="post">
        <input name="input_file" type="file">
        <input type="submit">
    </form>
</body>
</html>
```

Example **upload.html** in the folder templates:
```
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Uploaded file</title>
</head>
<body>
    <div>
        {{ file_name }}
    </div>
</body>
</html>
```
